dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java

```diff
@@ -1,226 +1,66 @@
-                    results.add(meterProvider.shutdown());
-                }
-
-                SdkLoggerProvider loggerProvider = sdk.getSdkLoggerProvider();
-                if (loggerProvider != null) {
-                    results.add(loggerProvider.shutdown());
-                }
-
-                CompletableResultCode.ofAll(results).join(10, TimeUnit.SECONDS);
-            }
-        } catch (Exception e) {
-            Tr.error(tc, Tr.formatMessage(tc, "CWMOT5002.telemetry.error", e));
-        }
-    }
-
-    private boolean checkDisabled(Map<String, String> oTelConfigs) {
-        //In order to enable any of the tracing aspects, the configuration otel.sdk.disabled=false must be specified in any of the configuration sources available via MicroProfile Config.
-        if (oTelConfigs.get(ENV_DISABLE_PROPERTY) != null) {
-            return Boolean.valueOf(oTelConfigs.get(ENV_DISABLE_PROPERTY));
-        } else if (oTelConfigs.get(CONFIG_DISABLE_PROPERTY) != null) {
-            return Boolean.valueOf(oTelConfigs.get(CONFIG_DISABLE_PROPERTY));
-        }
-        return true;
-    }
-
-    private HashMap<String, String> getTelemetryProperties() {
-        try {
-            HashMap<String, String> telemetryProperties = new HashMap<>();
-            for (String propertyName : config.getPropertyNames()) {
-                if (propertyName.startsWith("otel.")) {
-                    config.getOptionalValue(propertyName, String.class).ifPresent(
-                                                                                  value -> telemetryProperties.put(propertyName, value));
-                }
-            }
-            //Metrics and logs are disabled by default
-            telemetryProperties.put(CONFIG_METRICS_EXPORTER_PROPERTY, "none");
-            telemetryProperties.put(CONFIG_LOGS_EXPORTER_PROPERTY, "none");
-            telemetryProperties.put(ENV_METRICS_EXPORTER_PROPERTY, "none");
-            telemetryProperties.put(ENV_LOGS_EXPORTER_PROPERTY, "none");
-            return telemetryProperties;
-        } catch (Exception e) {
-            Tr.error(tc, Tr.formatMessage(tc, "CWMOT5002.telemetry.error", e));
-            return new HashMap<String, String>();
-        }
+    public OpenTelemetry getOpenTelemetry(OpenTelemetryInfo openTelemetryInfo) {
+        return super.getOpenTelemetry(openTelemetryInfo);
     }
 
+    @Override
     @Produces
     public Tracer getTracer(OpenTelemetry openTelemetry) {
-        try {
-            return openTelemetry.getTracer(INSTRUMENTATION_NAME);
-        } catch (Exception e) {
-            Tr.error(tc, Tr.formatMessage(tc, "CWMOT5002.telemetry.error", e));
-            return OpenTelemetry.noop().getTracerProvider().get("");
-        }
+        return super.getTracer(openTelemetry);
     }
 
+    @Override
     @Produces
     @ApplicationScoped
     public Span getSpan() {
-        return new SpanProxy();
+        return super.getSpan();
     }
```

Comment:
Why is this an override which then calls super?

Reply:
CDI producers are not inherited (if they were you'd get an ambigious resolution between the super and subclass). So those method exist just to be annotated.